### ISSUE TITLE
`Layout/LineLength` `AutoCorrect` is not working on the most recent versions

### ISSUE THREAD

**Issue by Andy9822** [rubocop/rubocop#12256]
## Expected behavior

I have the `Layout/LineLength` rule set up in my `.rubocop.yml` file.
I tried to set `AutoCorrect: true` but it doesn't auto-correct those lines, even the most basic ones. It simply doesn't autocorrect anything, only points out the lines that exceeded my `max`.

It should add a new line on my variable assignment so the string value that's being assigned goes to the next line and take fewer columns.

## Actual behavior

It only finds and points out the offenses but doesn't correct them:

```sh 
[15:49:07] ➜ bundle exec rubocop --autocorrect-all components/core/app/services/core/deals/create_service.rb
Inspecting 1 file
C

Offenses:

components/core/app/services/core/deals/create_service.rb:69:121: C: Layout/LineLength: Line is too long. [135/120]
              redis_key = "something_something_something_something_:#{@source.pipeline.account.id}:#{@source.id}:#{@inbound_mail.from}"
                                                                                                                        ^^^^^^^^^^^^^^^
```
<img width="1155" alt="image" src="https://github.com/rubocop/rubocop/assets/19315023/e92b79c1-9f24-413e-9951-899cb5cdd950">

## Steps to reproduce the problem

Add this to `.rubocop.yml`

```yaml
Layout/LineLength:
  Max: 120
  AutoCorrect: true
```

Try to run rubocop (i.e. `bundle exec rubocop --autocorrect-all <path>`)

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
[15:56:10] ➜ bundle exec rubocop -V
1.56.4 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 2.7.7) [arm64-darwin22]
  - rubocop-rails 2.21.1
```

**Comment by Andy9822** [comment]
I'm closing it because it started working all of a sudden when I tried it back yesterday so I assume it was something with my setup
